src/eduid_common/session/redis_session.py

```diff
@@ -296,241 +288,58 @@ def commit(self):
-        # The last byte ('x') is padding to prevent b32encode from adding an = at the end
-        combined = base64.b32encode(session_id + sig + b'x')
-        # Make sure token will be a valid NCName (pysaml2 requirement)
-        while combined.endswith(b'='):
-            combined = combined[:-1]
-        return (TOKEN_PREFIX + combined).decode('utf-8')
-
-    @staticmethod
-    def decode_token(token):
-        """
-        Decode a token (token is what is stored in a cookie) into it's components.
-
-        :param token: the token with the signed session_id
-        :type token: str | unicode
+        data = self.encrypt_data(self._data)
+        logger.debug(f'Committing session {self.token} to the cache with ttl {self.ttl} ({len(data)} bytes)')
+        self.conn.setex(self.token.session_id, self.ttl, data)
 
-        :return: the session_id and signature
-        :rtype: str | unicode, str | unicode
+    def encrypt_data(self, data_dict: Mapping[str, Any]) -> str:
         """
-        #  the slicing is to remove a leading 'a' needed so we have a
-        # valid NCName so pysaml2 doesn't complain when it uses the token as
-        # session id.
-        if not token.startswith(TOKEN_PREFIX):
-            raise ValueError('Invalid token string {!r}'.format(token))
-        val = token.strip(b'"')[len(TOKEN_PREFIX) :]
-        # Split the token into it's two parts - the session_id and the HMAC signature of it
-        # (the last byte is ignored - it is padding to make b32encode not put an = at the end)
-        _decoded = base64.b32decode(val)
-        _bin_session_id, _bin_sig = _decoded[:HMAC_DIGEST_SIZE], _decoded[HMAC_DIGEST_SIZE:-1]
-        return _bin_session_id, _bin_sig
-
-    def sign_data(self, data_dict):
-        """
-        Sign (and encrypt) data before storing it in Redis.
+        Sign and encrypt data before storing it in Redis.
 
         :param data_dict: Data to be stored
         :return: serialized data
-        :rtype: str | unicode
         """
-        if self.debug:
-            logger.debug('Storing data in cache[{}]:\n{!r}'.format(self.session_id, data_dict))
+        logger.debug(f'Storing data in cache[{self.token}]:\n{repr(data_dict)}')
         nonce = nacl.utils.random(nacl.secret.SecretBox.NONCE_SIZE)
         # Version data to make it easier to know how to decode it on reading
         data_json = json.dumps(data_dict, cls=NameIDEncoder)
         versioned = {
             'v2': bytes(
-                self.nacl_box.encrypt(data_json.encode('ascii'), nonce, encoder=nacl.encoding.Base64Encoder)
+                self.secret_box.encrypt(data_json.encode('ascii'), nonce, encoder=nacl.encoding.Base64Encoder)
             ).decode('ascii')
         }
         return json.dumps(versioned)
 
-    def verify_data(self, data_str):
+    def decrypt_data(self, data_str: str) -> Dict[str, Any]:
         """
         Verify (and decrypt) session data read from Redis.
```

Comment:
Decrypt and JSON load session data stored in Redis.

Fits better with encrypt data doc string above, "Sign and encrypt data before storing it in Redis."